plugins/producer_plugin/include/eosio/producer_plugin/unapplied_transaction_queue.hpp

```diff
@@ -0,0 +1,109 @@
+/**
+ *  @file
+ *  @copyright defined in eos/LICENSE
+ */
+
+#pragma once
+
+#include <eosio/chain/transaction_metadata.hpp>
+#include <eosio/chain/block_state.hpp>
+
+#include <queue>
+
+namespace eosio {
+
+/**
+ * Track unapplied transactions for aborted blocks, forked blocks, and subjectively
+ * failed transactions.
+ */
+class unapplied_transaction_queue {
```

Comment:
Previously the unapplied transactions were stored in a map keyed by transaction ID. So duplicates were prevented. Now it is possible for the `unapplied_transaction_queue` to have duplicates. I suppose this isn't necessarily a problem, but would using a `multi_index` in this implementation to enforce uniqueness be too much compared to the current lighter-weight implementation that uses only `deque`s?